openmp/libomptarget/src/OffloadRTL.cpp

```diff
@@ -48,7 +48,8 @@ void deinitRuntime() {
   std::scoped_lock<decltype(PluginMtx)> Lock(PluginMtx);
   assert(PM && "Runtime not initialized");
 
-  if (RefCount-- == 0) {
+  RefCount--;
```

Comment:
This looks wrong - you're separating the decrement from the load, should be a fetch_sub followed by a branch on the old value